src/network/mcpe/convert/TypeConverter.php

```diff
@@ -272,6 +273,10 @@ public function createInventoryAction(NetworkInventoryAction $action, Player $pl
 					}
 					$pSlot = $action->inventorySlot;
 
+					if($pSlot === UIInventorySlotOffset::BEACON_PAYMENT){
+						return new BeaconAction($old, $new);
```

Comment:
This is incorrectly implemented. The item is not directly consumed when it's placed into the beacon window, only when the effects have been selected. Therefore, the inventory should work the same as any other temporary inventory such as crafting tables.